homeassistant/components/litterrobot/config_flow.py

```diff
@@ -0,0 +1,51 @@
+"""Config flow for Litter-Robot integration."""
+import logging
+
+from pylitterbot.exceptions import LitterRobotException, LitterRobotLoginException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .const import DOMAIN  # pylint:disable=unused-import
+from .hub import LitterRobotHub
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Litter-Robot."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+
+        if user_input is not None:
+            for entry in self._async_current_entries():
+                if entry.data[CONF_USERNAME] == user_input[CONF_USERNAME]:
+                    return self.async_abort(reason="already_configured")
+
+            hub = LitterRobotHub(self.hass, user_input)
+            try:
+                await hub.login()
+                return self.async_create_entry(
```

Comment:
Please only wrap the line that can raise in the try... except block. Use an `else:` block if needed.